File: July/Day17/maximumLength.cc
#include <vector>
#include <algorithm>

class Solution {
public:
    int maximumLength(std::vector<int>& nums, int k) {
        std::vector<std::vector<int>> dp(k, std::vector<int>(k, 0));
        int maxSubsequence = 0;

        for (int num : nums) {
            num %= k;
            for (int prev = 0; prev < k; ++prev) {
                dp[prev][num] = dp[num][prev] + 1;
                maxSubsequence = std::max(maxSubsequence, dp[prev][num]);
            }
        }

        return maxSubsequence;
    }
};

File: July/Day17/maximumLength.js
function maximumLength(nums, k) {
    const dp = Array.from({ length: k }, () => Array(k).fill(0));
    let maxSubsequence = 0;

    for (let num of nums) {
        num = num % k;
        for (let prev = 0; prev < k; prev++) {
            dp[prev][num] = dp[num][prev] + 1;
            maxSubsequence = Math.max(maxSubsequence, dp[prev][num]);
        }
    }

    return maxSubsequence;
}
